common/src/app/common/pages/helpers.cljc

```diff
@@ -109,6 +109,20 @@
         (recur (conj result parent-id) parent-id)
         result))))
 
+(defn get-parent-ids-with-index
+  "Returns a tuple with the list of parents and a map with the position in everyparent"
+  [objects shape-id]
+  (loop [result [[] {}]
```

Comment:
split the result in two different bindings, this will avoid create an additional vector instance on each iteration

Reply:
good idea. Thanks